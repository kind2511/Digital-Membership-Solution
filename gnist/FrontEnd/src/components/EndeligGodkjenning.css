.eg-godkjenning-container {
  background-color: #f8f8f8;
  border: 1px solid #ddd;
  border-radius: 8px;
  padding: 20px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  overflow: hidden;
}

.eg-godkjenning-container .eg-section-title {
  color: #333;
  font-size: 20px;
  margin-bottom: 10px;
}

.eg-section-content {
  max-height: 200px;
  overflow-y: auto;
}

.eg-members-list {
  list-style: none;
  padding: 0;
  margin: 0;
}

.eg-members-list li {
  padding: 10px;
  background-color: #fff;
  border: 1px solid #ddd;
  margin-top: 5px;
  cursor: pointer;
  transition: background-color 0.3s;
}

.eg-members-list li:hover {
  background-color: #e9e9e9;
}

.eg-modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
}

.eg-modal-content {
  background-color: #fff;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  width: 40%;
  min-width: 300px;
}

.eg-modal-content p {
  font-size: 16px;
  color: #333;
  margin-top: 10px;
}

.eg-modal-buttons {
  display: flex;
  justify-content: space-between;
  margin-top: 20px;
}

.eg-button-lukk {
  padding: 8px 16px;
  background-color: red;
  color: white;
  border: none;
  border-radius: 5px;
  cursor: pointer;
}

.eg-button-godkjent {
  padding: 8px 16px;
  background-color: green;
  color: white;
  border: none;
  border-radius: 5px;
  cursor: pointer;
}
.eg-button-ikke-godkjent {
  padding: 8px 16px;
  background-color: #dc3545; /* Red color */
  color: white;
  border: none;
  border-radius: 5px;
  cursor: pointer;
}

/* New Success message banner */
.verification-success-banner {
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #28a745; 
  color: white;
  text-align: center;
  padding: 10px;
  font-weight: bold;
  z-index: 1500;
  animation: slideUp 0.5s ease-in-out;
}

/* Keyframes for the banner slide-up effect */
@keyframes slideUp {
  from {
      bottom: -50px;
      opacity: 0;
  }
  to {
      bottom: 0;
      opacity: 1;
  }
}