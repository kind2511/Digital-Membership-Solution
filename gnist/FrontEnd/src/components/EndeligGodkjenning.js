import React, { useEffect, useState } from 'react';
import './EndeligGodkjenning.css';

function EndeligGodkjenning() {
    const [unverifiedMembers, setUnverifiedMembers] = useState([]);
    const [selectedMember, setSelectedMember] = useState(null);
    const [showVerificationSuccessMessage, setShowVerificationSuccessMessage] = useState(false);

    useEffect(() => {
        fetchUnverifiedMembers();
    }, []);

    const fetchUnverifiedMembers = async () => {
        try {
            const response = await fetch('http://127.0.0.1:8000/digital_medlemsordning/get_all_unverified_members/');
            if (!response.ok) {
                throw new Error('Failed to fetch unverified members');
            }
            const data = await response.json();
            setUnverifiedMembers(data);
        } catch (error) {
            console.error(error);
        }
    };

    const verifyMember = async (member) => {
        try {
            const response = await fetch(`http://127.0.0.1:8000/digital_medlemsordning/verify_member/${member.auth0ID}/`, {
                method: 'PUT',
                headers: {
                    'Content-Type': 'application/json',
                },
            });
            if (!response.ok) {
                throw new Error('Failed to verify member');
            }
            setShowVerificationSuccessMessage(true);
            setTimeout(() => setShowVerificationSuccessMessage(false), 3000);
            fetchUnverifiedMembers();
            handleClose();
        } catch (error) {
            console.error(error);
        }
    };

    const handleClose = () => {
        setSelectedMember(null);
    };

    return (
        <div className="eg-godkjenning-container">
            <h2 className="eg-section-title">Endelig Godkjenning</h2>
            <div className="eg-section-content">
                <ul className="eg-members-list">
                    {unverifiedMembers.map(member => (
                        <li key={member.userID} onClick={() => setSelectedMember(member)}>
                            {member.first_name} {member.last_name}
                        </li>
                    ))}
                </ul>
            </div>
            {selectedMember && (
                <div className="eg-modal-overlay">
                    <div className="eg-modal-content">
                        <p><strong>Navn:</strong> {selectedMember.first_name} {selectedMember.last_name}</p>
                        <p><strong>Navn og Tlf for Foresatt:</strong> {selectedMember.guardian_name || 'N/A'} {selectedMember.guardian_phone || 'N/A'}</p>
                        <div className="eg-modal-buttons">
                            <button className="eg-button-lukk" onClick={handleClose}>Lukk</button>
                            <button className="eg-button-godkjent" onClick={() => verifyMember(selectedMember)}>Godkjent</button>
                        </div>
                    </div>
                </div>
            )}
            {showVerificationSuccessMessage && (
                <div className="verification-success-banner">
                    Medlem ble verifisert
                </div>
            )}
        </div>
    );
}

export default EndeligGodkjenning;
