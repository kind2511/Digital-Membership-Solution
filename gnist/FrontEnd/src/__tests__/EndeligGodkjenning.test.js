import React from 'react';
import renderer from 'react-test-renderer';
import EndeligGodkjenning from '../components/EndeligGodkjenning';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Mock axios for the test
const mock = new MockAdapter(axios);
const unverifiedMembersMock = [
  {
    userID: 1,
    auth0ID: 'auth0|123456789',
    first_name: 'John',
    last_name: 'Doe',
    birthdate: '2000-01-01',
    guardian_name: 'Jane Doe',
    guardian_phone: '12345678'
  },
  {
    userID: 2,
    auth0ID: 'auth0|987654321',
    first_name: 'Jane',
    last_name: 'Doe',
    birthdate: '2002-02-02',
    guardian_name: null,
    guardian_phone: null
  }
];
mock.onGet('http://127.0.0.1:8000/digital_medlemsordning/get_all_unverified_members/').reply(200, unverifiedMembersMock);

test('EndeligGodkjenning component snapshot', () => {
  const component = renderer.create(<EndeligGodkjenning />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
