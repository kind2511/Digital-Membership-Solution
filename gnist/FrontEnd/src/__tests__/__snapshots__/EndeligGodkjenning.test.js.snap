// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndeligGodkjenning component snapshot 1`] = `
<div
  className="eg-godkjenning-container"
>
  <h2
    className="eg-section-title"
  >
    Endelig Godkjenning
  </h2>
  <div
    className="eg-section-content"
  >
    <ul
      className="eg-members-list"
    />
  </div>
</div>
`;
